c++/src/sargs/SargsApplier.cc

```diff
@@ -25,7 +26,13 @@ namespace orc {
                                     const std::string& colName) {
     for (uint64_t i = 0; i != type.getSubtypeCount(); ++i) {
       if (type.getFieldName(i) == colName) {
-        return type.getSubtype(i)->getColumnId();
+        if (type.getKind() == CHAR || type.getKind() == VARCHAR) {
+          ///FIXME: disable PPD for char and varchar types as their rules
+          // vary among different engines.
+          return INVALID_COLUMN_ID;
```

Comment:
Do we have a test coverage for this code path?
If not, I'm +1 for @pgaref 's opinion. We can do this later.